File: Examples/runtimes/java/DynamoDbEncryption/src/test/java/software/amazon/cryptography/examples/TestGetEncryptedDataKeyDescriptionExample.java
package software.amazon.cryptography.examples;

import java.util.UUID;
import org.junitpioneer.jupiter.RetryingTest;

public class TestGetEncryptedDataKeyDescriptionExample {

  @RetryingTest(3)
  public void TestGetEncryptedDataKeyDescription() {
    final String partitionKeyValue =
      "GetEncryptedDataKeyDescriptionExample" + UUID.randomUUID();
    final String partitionKeyName = "partition_key";
    final String sortKeyValue = "0";
    final String sortKeyName = "sort_key";
    // `getEncryptedDataKeyDescription` assumes that there is already existing item in dynamodb.
    // So, we need to run `PutItemGetItem` before `getEncryptedDataKeyDescription`
    BasicPutGetExample.PutItemGetItem(
      TestUtils.TEST_KMS_KEY_ID,
      TestUtils.TEST_DDB_TABLE_NAME,
      partitionKeyName,
      sortKeyName,
      partitionKeyValue,
      sortKeyValue
    );
    GetEncryptedDataKeyDescriptionExample.getEncryptedDataKeyDescription(
      TestUtils.TEST_DDB_TABLE_NAME,
      partitionKeyName,
      partitionKeyValue,
      sortKeyName,
      sortKeyValue,
      "aws-kms",
      TestUtils.TEST_KMS_KEY_ID,
      null,
      null
    );
    TestUtils.cleanUpDDBItem(
      TestUtils.TEST_DDB_TABLE_NAME,
      partitionKeyName,
      sortKeyName,
      partitionKeyValue,
      sortKeyValue
    );
  }
}

File: Examples/runtimes/python/DynamoDBEncryption/test/test_get_encrypted_data_key_description_example.py
# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0
"""Test get encrypted data key description example."""
import pytest

from ..src.get_encrypted_data_key_description_example import get_encrypted_data_key_description
from .test_utils import (
    TEST_DDB_TABLE_NAME,
    TEST_KMS_KEY_ID,
)

pytestmark = [pytest.mark.examples]


def test_get_encrypted_data_key_description():
    """Test get encrypted data key description example."""
    get_encrypted_data_key_description(
        TEST_DDB_TABLE_NAME,
        "partition_key",
        "BasicPutGetExample",
        "sort_key",
        "0",
        "aws-kms",
        TEST_KMS_KEY_ID,
        None,
        None,
    )
